### ISSUE TITLE
Error: The time zone ID 'Unknown' was not found on the local computer when running on Azure

### ISSUE THREAD

**Issue by imsam67** [aspnet/SignalR#1296]
I've implemented SignalR Core (alpha 2) on an ASP.NET Core 2.0 app with React front-end.

Everything seems to be working fine locally. I'm able to send messages (in a chat) back and forth between Chrome and Edge.

I then published my app to Azure and enabled web sockets in App Settings on Azure. When I try to send a message from Chrome, I get an error that reads:

> Uncaught (in promise) Error: The time zone ID 'Unknown' was not found on the local computer.
![signalr_error](https://user-images.githubusercontent.com/15844030/34889696-f9bf1bf8-f79c-11e7-8083-f26c310a2195.PNG)

This message never gets through to Edge. If I send a message from Edge to Chrome, it seems to go through and Chrome picks it up right away.

One more important point: even though the message sent from Chrome doesn't seem to get through and Edge doesn't pick it up, if I refresh the page on Edge, I actually see the message Chrome sent. This means, the message sent by Chrome does reach the hub and get recorded in my database.

Any idea what may be causing this?

**Comment by BrennanConroy** [comment]
Looks like the error might be coming from an exception in your Hub, I'm guessing you're doing some time zone stuff there?

**Comment by imsam67** [comment]
@BrennanConroy  Yes, I'm doing some time zone conversions. What's interesting is that this works locally but when I run it on Azure, I'm getting System.TimeZoneNotFoundException error. I simply call a simple util method that converts UTC to user's time zone. Here's what my util method looks like:

```
public static DateTime ConvertFromUtc(DateTime input, Enums.General.TimeZone timeZone)
{
    var targetTimeZone = Enum.EnumUtils.GetDescription(timeZone);
    var targetTime = TimeZoneInfo.ConvertTime(input, TimeZoneInfo.Utc, TimeZoneInfo.FindSystemTimeZoneById(targetTimeZone));
    return targetTime;
}
```
I simply pass a DateTime value which is in UTC and an enum value that gets converted to TimeZone Id in .NET.

Here's what my enum looks like:
```
public enum TimeZone
{
   [Description("Pacific Standard Time")]
   PacificStandardTime = 6,
   [Description("US Mountain Standard Time")]
   USMountainStandardTime = 7,
   [Description("Mountain Standard Time (Mexico)")]
   MountainStandardTimeMexico = 8,
   [Description("Mountain Standard Time")]
   MountainStandardTime = 9,
   // ... The rest is omitted for brevity
}
```
Any idea why this would not work on Azure App service but work fine locally. Also, I call this util method in other places in my ASP.NET Core 2.0 app and it works fine even on Azure. Looks like it's only throwing an error in the SignalR hub.

**Comment by BrennanConroy** [comment]
Where is the DateTime coming from?
What protocol are you using?

**Comment by imsam67** [comment]
Here's what's happening. I receive a message object in my hub which contains the message the user has types along with user info which includes his/her time zone Id. This is Id is coming from my enum. For example 9 for MST.

I then create the message in the database and get an updated set of messages for the chat the user is participating in. All date/time values are kept in UTC so before I send the updated messages back to the user, I call my util method above to convert UTC values to user's time zone. This is exactly where it's blowing up. Looks like somehow the time zone is not being resolved correctly. See the screen shot from the Azure portal showing the exception.
![capture2](https://user-images.githubusercontent.com/15844030/34901116-5b312dea-f7d4-11e7-9593-eb9eaad4e4ee.PNG)


As per protocol, in my console, I'm seeing json -- see below:
![capture](https://user-images.githubusercontent.com/15844030/34901038-4e4ad03c-f7d3-11e7-8e5a-d9731aad34a4.PNG)

**Comment by BrennanConroy** [comment]
>Also, I call this util method in other places in my ASP.NET Core 2.0 app and it works fine even on Azure. Looks like it's only throwing an error in the SignalR hub.

Can you figure out what is different between calling the function in these 2 places?

**Comment by imsam67** [comment]
I changed my approach and the "symptoms" disappeared. Simply switched to using UTC offset to convert date/time from UTC to a particular time zone.

I saw another issue posted here that has to do with date/time. Not sure if they're related but just wanted to point it out.

For whatever it's worth: here's that issue: https://github.com/aspnet/SignalR/issues/1346

Closing this issue at this time. Thanks!
